### PR TITLE
ci: reenable goerr113 and unused linters across the codebase

### PR DESCRIPTION
**Pull request by ti-mo**
Does what it says on the tin. Now the `privileged_tests` tags in test files are gone, the unused linter can be useful again. Remove all code it flagged.

Two false positives were spotted: functions that are only used in `Benchmark___` functions. There have been a few bugs around this with `Example_` and `Test_`, but since it's just a few funcs for now, just `//nolint:unused` them.

```release-note
ci: reenable goerr113 and unused linters across the codebase
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
Thanks :100:

File: test/k8s/service_helpers.go
Comment (joestringer) [inline test/k8s/service_helpers.go:903]: @cilium/sig-datapath do we have coverage of this now? I'm surprised this was unused.

File: daemon/cmd/kube_proxy_replacement_test.go
Comment (joestringer) [inline daemon/cmd/kube_proxy_replacement_test.go:15]: Given the naming, I presume this was supposed to store & restore the KPR config so that global state doesn't impact later test runs? Should we just implement the restore piece rather than dropping this?

**Review by joestringer — COMMENTED** [review]
LGTM from my codeowners, just a couple of comments where maybe the right answer is to add the piece of logic that should have been using this code rather than deleting the code.

**Review by christarazi — APPROVED** [review]
LGTM pending Joe's comments

File: daemon/cmd/kube_proxy_replacement_test.go
Comment (joamaki) [reply daemon/cmd/kube_proxy_replacement_test.go:15]: This was a left over from earlier refactoring which separated this out from daemon_test.go (e.g. no longer relied on TestMain there that set the environment up). This test now calls `option.Config.Populate` and does not rely on it being in some reliable state. Other tests need to do the same as well, so we can drop the restoring.

File: test/k8s/service_helpers.go
Comment (joamaki) [reply test/k8s/service_helpers.go:903]: Looks like this wasn't cleaned up as part of the tests. We now have `test/controlplane/services/nodeport` testing that we populate lbmap correctly for HostPort and https://github.com/cilium/cilium-cli/blob/master/connectivity/tests/host.go tests e2e, so we should be good.

**Comment by ti-mo** [comment]
All relevant tests passing, merging.

**Pull request merged**
